src/controllers/queryRunner.ts

```diff
@@ -1365,6 +1454,168 @@ export default class QueryRunner {
+            });
+
+            let values: string[] = [];
+            let columnIndex = 0;
+
+            for (let i = 0; i < rowSelections.length; i++) {
+                let rowSelection = rowSelections[i];
+
+                // Add NULL values for gaps before this selection
+                while (columnIndex < rowSelection.fromCell - firstColumn) {
+                    values.push("NULL");
+                    columnIndex++;
+                }
+
+                let cellObjects = row.slice(rowSelection.fromCell, rowSelection.toCell + 1);
+
+                for (let cellObject of cellObjects) {
+                    if (cellObject.isNull) {
+                        values.push("NULL");
+                    } else {
+                        let displayValue = this.shouldRemoveNewLines()
+                            ? this.removeNewLines(cellObject.displayValue)
+                            : cellObject.displayValue;
+
+                        // Check if the value is numeric
+                        if (
+                            !isNaN(Number(displayValue)) &&
+                            isFinite(Number(displayValue)) &&
+                            displayValue.trim() !== ""
+                        ) {
+                            values.push(displayValue);
+                        } else {
+                            // Escape single quotes and wrap in single quotes for string values
+                            let escapedValue = displayValue.replace(/'/g, "''");
+                            values.push(`'${escapedValue}'`);
+                        }
+                    }
+                    columnIndex++;
+                }
+            }
+
+            // Add NULL values for gaps after the last selection
+            while (columnIndex < columnHeaders.length) {
+                values.push("NULL");
+                columnIndex++;
+            }
+
+            allRowValues.push(values);
+        }
+
+        // SQL Server limit: 1000 rows per INSERT VALUES statement
+        const maxRowsPerInsert = 1000;
+
+        if (allRowValues.length <= maxRowsPerInsert) {
+            // Use single INSERT INTO ... VALUES statement
+            const valueRows = allRowValues.map((values) => `    (${values.join(", ")})`);
+            return `INSERT INTO [TableName] (${columnList})\nVALUES\n${valueRows.join(",\n")};`;
+        } else {
+            // Break into individual INSERT statements
+            const insertStatements: string[] = [];
+
+            for (let i = 0; i < allRowValues.length; i += maxRowsPerInsert) {
+                const chunk = allRowValues.slice(i, i + maxRowsPerInsert);
+                const valueRows = chunk.map((values) => `    (${values.join(", ")})`);
+                insertStatements.push(
+                    `INSERT INTO [TableName] (${columnList})\nVALUES\n${valueRows.join(",\n")};`,
+                );
+            }
+
+            return insertStatements.join("\n\n");
```

Comment:
We can keep the "\n" like this or do we need to check the platform to use "\r\n"?